### DISCUSSION TITLE
Serialization of Page class

### DISCUSSION THREAD

**Discussion by luccaportes** [datapane/datapane#370] [Ideas]
Hi,

I was wondering if there is any way that we can serialize the Page class, That way we can parallelize the report creation on multiple steps of a pipeline.

We tried simply pickling the class, it saves and loads fine, but when we try to combine multiple pickles of pages in a report on a different machine, we got the following error:

`FileNotFoundError: [Errno 2] No such file or directory: '/tmp/dp-tmp-tat6pme_/dp-tmp-vaa7vafv.tbl.html'
`
This makes clear that some classes are not self-sufficient, as they have to resort to temporary HTML pages, which makes it impossible to serialize the way we are doing it.

Is there any way to do that?

**Comment by mands** [comment]
Hi @luccaportes ,

I've moved this to a Idea in our Discussions. We're looking at ways to support serialising View fragments that can then be loaded and inlined within larger Views / Reports - hope to have something to solve this in the next few weeks.
